pyquil/operator_estimation.py

```diff
@@ -479,3 +499,166 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
                 expectation=np.asscalar(obs_mean),
                 stddev=np.asscalar(np.sqrt(obs_var)),
             )
+
+
+def terms_diagonal_in_tpb(pauli_a: PauliTerm, pauli_b: PauliTerm):
```

Comment:
contrast with `_ops_diagonal_in_tpb`?

Reply:
They actually achieve the same effect. `_all_qubits_diagonal_in_tpb` acts on the union of qubits of the two operators, whereas `terms_diagonal_in_tpb` acts on the intersection of those qubits. I'm getting rid of the latter (since the former already exists in master), but (a) I'm moving the more extensive doc-string for `terms_diagonal_in_tpb` into that of `_all_qubits_diagonal_in_tpb`, and (b) changing the union of qubits to intersection of qubits in `_all_qubits_diagonal_in_tpb` (this seems safer to me against any potential future changes to the `PauliTerm` class).